docs/Client.md

```diff
@@ -0,0 +1,883 @@
+
+#### Parameter: `UpgradeData`
+
+* **headers** `http.IncomingHeaders`
+* **socket** `stream.Duplex`
+* **opaque** `unknown`
+
+#### Example 1 - Basic Upgrade Request
+
+```js
+'use strict'
+const { Client } = require('undici')
+const { createServer } = require('http')
+
+const server = createServer((request, response) => {
+  response.statusCode = 101
+  response.setHeader('connection', 'upgrade')
+  response.setHeader('upgrade', request.headers.upgrade)
+  response.end()
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  client
+    .upgrade({ path: '/' })
+    .then(({ headers, socket }) => {
+      socket.on('end', () => {
+        console.log(`upgrade: ${headers.upgrade}`)
+        client.close()
+        server.close()
+      })
+      socket.end()
+    })
+    .catch(error => {
+      console.error(error)
+      client.close()
+      server.close()
+    })
+})
+```
+
+## Instance Properties
+
+### `Client.busy`
+
+* `boolean`
+
+`true` if pipeline is saturated or blocked. Indicates whether dispatching further requests is meaningful.
+
+### `Client.closed`
+
+* `boolean`
+
+`true` after `client.close()` has been called.
+
+### `Client.connected`
+
+* `boolean`
+
+`true` if the client has an active connection. The client will lazily create a connection when it receives a request and will destroy it if there is no activity for the duration of the `timeout` value.
+
+### `Client.destroyed`
+
+* `boolean`
+
+`true` after `client.destroyed()` has been called or `client.close()` has been called and the client shutdown has completed.
+
+### `Client.pending`
+
+* `number`
+
+Number of queued requests.
+
+### `Client.pipelining`
+
+* `number`
+
+Property to get and set the pipelining factor.
+
+### `Client.running`
+
+* `number`
+
+Number of inflight requests.
+
+### `Client.size`
+
+* `number`
+
+Number of pending and running requests.
+
+## Instance Events
+
+### Event: `'connect'`
+
+Emitted when a socket has been created and connected. The client will connect once `Client.size > 0`.
+
+### Event: `'disconnect'`
+
+Returns:
+
+* **error** `Error`
+
+Emitted when socket has disconnected. The first argument of the event is the error which caused the socket to disconnect. The client will reconnect if or once `Client.size > 0`.
+
+### Event: `'drain'`
+
+Emitted when pipeline is no longer fully saturated.
+
+## Parameter: `UndiciHeaders`
+
+* `http.IncomingHttpHeaders | string[] | null`
+
+Header arguments such as `options.headers` in [`Client.dispatch`](./Client.md#client-dispatchoptions-handlers) can be specified in two forms; either as an object specified by the `http.IncomingHttpHeaders` type, or an array of strings. An array representation of a header list must have an even length or an `InvalidArgumentError` will be thrown.
+
+Keys are lowercase and values are not modified. If you don't specify a `host` header, it will be derived from the `url` of the [Client](#class-client) instance.
```

Comment:
This is not true for dispatch nor request headers (only response headers)

Reply:
This line was copied from https://github.com/nodejs/undici#clientrequestopts-callbackerr-data-promisevoid - i'll remove it now